crates/forge/tests/integration/get_class_hash.rs

```diff
@@ -0,0 +1,108 @@
+use crate::integration::common::corelib::{corelib_path, predeployed_contracts};
+use crate::integration::common::runner::Contract;
+use crate::{assert_passed, test_case};
+use camino::Utf8PathBuf;
+use forge::run;
+use indoc::indoc;
+use std::path::Path;
+
+#[test]
+fn get_class_hash() {
+    let test = test_case!(
+        indoc!(
+            r#"
+            use array::ArrayTrait;
+            use result::ResultTrait;
+            use snforge_std::{ declare, PreparedContract, deploy, start_prank, get_class_hash };
+
+            #[test]
+            fn test_get_class_hash() {
+                let class_hash = declare('GetClassHashCheckerUpg');
+                let prepared = PreparedContract { class_hash: class_hash, constructor_calldata: @ArrayTrait::new() };
+                let contract_address = deploy(prepared).unwrap();
+                assert(get_class_hash(contract_address) == class_hash, 'Wrong class hash');
+            }
+        "#
+        ),
+        Contract::from_code_path(
+            "GetClassHashCheckerUpg".to_string(),
+            Path::new("tests/data/contracts/get_class_hash_checker.cairo"),
+        )
+        .unwrap()
+    );
+
+    let result = run(
+        &test.path().unwrap(),
+        &String::from("src"),
+        &test.path().unwrap().join("src/lib.cairo"),
+        &Some(test.linked_libraries()),
+        &Default::default(),
+        &corelib_path(),
+        &test.contracts(&corelib_path()).unwrap(),
+        &Utf8PathBuf::from_path_buf(predeployed_contracts().to_path_buf()).unwrap(),
+    )
+    .unwrap();
+
+    assert_passed!(result);
+}
+
+#[test]
+fn get_class_hash_replace_class() {
+    let test = test_case!(
+        indoc!(
+            r#"
+            use array::{ArrayTrait, SpanTrait};
+            use core::result::ResultTrait;
+            use starknet::ClassHash;
+            use snforge_std::{declare, deploy, get_class_hash, PreparedContract};
+
+            #[starknet::interface]
+            trait IUpgradeable<T> {
+                fn upgrade(ref self: T, class_hash: ClassHash);
+            }
+
+            #[test]
+            fn test_get_class_hash_replace_class() {
+                let class_hash = declare('GetClassHashCheckerUpg');
+
+                let prepared = PreparedContract {
+                    class_hash: class_hash,
+                    constructor_calldata: @ArrayTrait::new()
+                };
+
+                let contract_address = deploy(prepared).unwrap();
+
+                assert(get_class_hash(contract_address) == class_hash, 'Bad class hash');
+
+                let hsn_class_hash = declare('HelloStarknet');
+                IUpgradeableDispatcher { contract_address }.upgrade(hsn_class_hash);
+                assert(get_class_hash(contract_address) == hsn_class_hash, 'Bad class hash upg');
```

Comment:
Can we try to call a function from `HelloStarknet` after upgrade?

Reply:
That's a very nice idea. :+1: